modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationListener.java

```diff
@@ -5,21 +5,42 @@
  */
 package com.opengamma.strata.calc.runner;
 
+import java.util.List;
+
 import com.opengamma.strata.basics.CalculationTarget;
 import com.opengamma.strata.calc.CalculationRunner;
+import com.opengamma.strata.calc.Column;
 
 /**
  * Listener that is notified when calculations are performed by a {@link CalculationRunner}.
  * <p>
  * It is guaranteed that the methods of a listener will only be invoked by a single thread at any
- * time. Therefore listener implementations are not necessarily required to be thread safe.
+ * time. It is not guaranteed to be the same thread invoking a listener each time. The calling
+ * code is synchronized to ensure that any changes in the listener state will be
+ * visible to every thread used to invoke the listener. Therefore listener implementations
+ * are not required to be thread safe.
  * <p>
- * It is not guaranteed to be the same thread invoking a listener each time.
+ * A listener instance should not be used for multiple sets of calculations.
  */
 public interface CalculationListener {
 
   /**
-   * Invoked when a calculation completes.
+   * Invoked when the calculations start; guaranteed to be invoked
+   * before {@link #resultReceived(CalculationTarget, CalculationResult)} and
+   * {@link #calculationsComplete()}.
+   *
+   * @param columns the columns for which values are being calculated
+   */
+  public default void calculationsStarted(List<Column> columns) {
```

Comment:
Seems reasonable. Our implementations won't use it but there's no harm adding it and I can see why it might be useful. It's also available at the call site so it's easy to add.